include/memory/memory.hh

```diff
@@ -3,27 +3,30 @@
 
 #include "common/common.hh"
 #include <iostream>
+#include <list>
+#include <memory>
 #include <unordered_map>
+#include <algorithm>
+#include <array>
 
 namespace sim {
 
+class PhysMemory;
+
 class Memory final {
 public:
   struct MemoryStats {
 
     std::size_t numLoads{};
     std::size_t numStores{};
-    std::size_t numPageFaults{};
   };
 
 private:
-  std::unordered_map<Addr, Word> mem{};
   MemoryStats stats{};
-
-  Word pageFaultHandle(Addr addr);
+  std::unique_ptr<PhysMemory> physMem{};
```

Comment:
Why do you need a pointer here?

Reply:
Thought about storing PhysMemory on stack could cause stack overflow